### ISSUE TITLE
Enable @typescript-eslint/no-shadow and @typescript-eslint/no-redeclare for the project

### ISSUE THREAD

**Issue by benjdlambert** [backstage/backstage#4219]
## Feature Suggestion
Now after #4210 is merged, we should enabled the two following rules in our `cli/config/eslint*.js` files.

```js
{
  rules: {
    '@typescript-eslint/no-shadow': 'error',
    '@typescript-eslint/no-redeclare': 'error'
  }
}
```

And then update the project to work with these new rules :)

**Comment by ron-debajyoti** [comment]
Since this seems easy for a beginner, I would like to give it a try.

**Comment by freben** [comment]
Nice, thanks!

**Comment by ron-debajyoti** [comment]
Does updating the project also includes fixing the eslint errors based on the new rules, for all the packages and plugins of backstage ?

**Comment by freben** [comment]
Yeah. We postponed enabling these rules specifically because there are some places where things need to be fixed accordingly.

**Comment by ron-debajyoti** [comment]
I have been fixing the `no-shadow` and `no-redeclare` errors caused by changing the appropriate variable names. Many files have such errors although the changes in each file are 1-2 lines. So far, I have changes in 32 files, and currently fixing the errors in  `plugin/cost-insight`.

Is it advisable to make all the changes and make the commit or do I commit in batches for easy review ?

**Comment by freben** [comment]
Sounds like it's fine to continue doing them all as one batch.

**Comment by ron-debajyoti** [comment]
I made the pull request but I am confused about what should be the correct changes in two files, which are causing tests to fail. I mentioned it in the pull request and would appreciate the feedback.
